coordinator/internal/transitengineapi/transitengineapi.go

```diff
@@ -253,3 +288,40 @@ func loggingMiddleware(next http.HandlerFunc, logger *slog.Logger) http.HandlerF
 		}
 	})
 }
+
+func authorizationMiddleware(next http.HandlerFunc) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		workloadSecretID := r.PathValue("name")
+		if err := authorizeWorkloadSecret(workloadSecretID, r); err != nil {
+			http.Error(w, fmt.Sprintf("Unauthorized: %v", err), http.StatusUnauthorized)
```

Comment:
`http.StatusUnauthorized` is only correct if the client is not _authenticated_ (see https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Status/401). Since we authenticate at the TLS layer, we can always return 403 here.